### PR TITLE
[SPMD] Use xs.Mesh in test_2d_tensor_3d_mesh

### PR DESCRIPTION
**Pull request by khatwanimohit**
Use `Mesh` in the test for 2d_tensor_3d_mesh until V2 support is added in `HybridMesh`

### CODE REVIEW

File: test/spmd/test_xla_sharding.py
Comment (jonb377) [inline test/spmd/test_xla_sharding.py:485]: Since we're not using HybridMesh, we can remove the mocking logic here

**Review by jonb377 — APPROVED** [review]
Thanks Mohit!

File: test/spmd/test_xla_sharding.py
Comment (alanwaketan) [reply test/spmd/test_xla_sharding.py:491]: Saw that it's covered by `test_hybrid_mesh_shape`.

**Review by alanwaketan — APPROVED** [review]

**Comment by JackCaoG** [comment]
waiting for TPU CI before merge.

**Pull request merged**
